### PR TITLE
cleanup makefile

### PR DESCRIPTION
**Pull request by adrianchiris**
- cleanup makefile
- update workflows
- remove deprecated linters from .golangci.yml
- allow to provide additional GO build options, ldflags and flags

### CODE REVIEW

**Comment by adrianchiris** [comment]
@SchSeba @zeeke PTAL mellanox ci failure is un-related

File: Makefile [hunk 1]
```diff
@@ -42,125 +26,85 @@ ifdef HTTPS_PROXY
...
-$(BUILDDIR)/$(BINARY_NAME): $(GOFILES) | $(BUILDDIR)
-	@cd $(BASE)/cmd/$(BINARY_NAME) && CGO_ENABLED=0 go build $(LDFLAGS) -o $(BUILDDIR)/$(BINARY_NAME) -tags no_openssl -v
-
-$(GOLINT): | $(BASE) ; $(info  building golint...)
-	$(call go-install-tool,$(GOLINT),golang.org/x/lint/golint@latest)
-
-$(GOCOVMERGE): | $(BASE) ; $(info  building gocovmerge...)
-	$(call go-install-tool,$(GOCOVMERGE),github.com/wadey/gocovmerge@latest)
+GOLANGCI_LINT = $(BINDIR)/golangci-lint
+GOLANGCI_LINT_VERSION ?= v1.55.2
```
Comment (Eoghan1232) [inline Makefile:60]: should we keep variables at the top of Makefile for ease?

Comment (adrianchiris) [reply Makefile:60]: my preference is to keep those next to the relevant target thats downloading it :) it feels clearer to me.
a change in fashion (as in the past i was for keeping it at the top :D)

File: Makefile [hunk 2]
```diff
@@ -42,125 +26,85 @@ ifdef HTTPS_PROXY
...
+test-coverage: | $(COVERAGE_DIR) ; $(info  Running coverage tests...) @ ## Run coverage tests
+	$Q go test -timeout 30s -cover -covermode=$(COVERAGE_MODE) -coverprofile=$(COVERAGE_PROFILE) $(PKGS)
 
-.PHONY: fmt
-fmt: ; $(info  running gofmt...) @ ## Run gofmt on all source files
-	@ret=0 && for d in $$(go list -f '{{.Dir}}' ./...); do \
-		gofmt -l -w $$d/*.go || ret=$$? ; \
-	 done ; exit $$ret
+.PHONY: lint
+lint: $(GOLANGCI_LINT) ## Run golangci-lint linter
```
Comment (Eoghan1232) [inline Makefile:85]:
```suggestion
lint: $(GOLANGCI_LINT) ; $(info Running golangci-lint linter...) @ ## Run golangci-lint linter
```

File: Makefile [hunk 3]
```diff
@@ -42,125 +26,85 @@ ifdef HTTPS_PROXY
...
 
-$(GOBIN):
-	@mkdir -p $@
+$(BUILDDIR): ; $(info Creating build directory...)
+	$Q mkdir -p $@
 
-$(BUILDDIR): | $(BASE) ; $(info Creating build directory...)
-	@cd $(BASE) && mkdir -p $@
+$(COVERAGE_DIR): ; $(info Creating coverage directory...)
+	$Q mkdir -p $@
```
Comment (Eoghan1232) [inline Makefile:53]:
```
.PHONY: create-dirs
create-dirs: $(BINDIR) $(BUILDDIR) $(COVERAGE_DIR)

$(BINDIR) $(BUILDDIR) $(COVERAGE_DIR):
	$Q mkdir -p $@

```

File: Makefile [hunk 4]
```diff
@@ -42,125 +26,85 @@ ifdef HTTPS_PROXY
...
 
-build: $(BUILDDIR)/$(BINARY_NAME) | ; $(info Building $(BINARY_NAME)...) @ ## Build SR-IOV Network device plugin
+.PHONY: build
+build: | $(BUILDDIR) ; $(info Building $(BINARY_NAME)...) @ ## Build SR-IOV Network device plugin
```
Comment (Eoghan1232) [inline Makefile:55]:
```suggestion
build: create-dirs ;  $(BUILDDIR) ; $(info Building $(BINARY_NAME)...) @ ## Build SR-IOV Network device plugin
```

File: Makefile (hunk 1 above)
Comment (Eoghan1232) [reply Makefile:60]: Makes sense :)

File: Makefile (hunk 3 above)
Comment (Eoghan1232) [reply Makefile:53]: not sure if needed, but might be a nice way to have it...

File: Makefile (hunk 4 above)
Comment (Eoghan1232) [reply Makefile:55]: if you go with the previous suggestion...

Comment (adrianchiris) [reply Makefile:55]: we depend only on build dir here.

File: Makefile (hunk 3 above)
Comment (adrianchiris) [reply Makefile:53]: ack, will modify

File: Makefile (hunk 2 above)
Comment (adrianchiris) [reply Makefile:85]: ack

**Review by Eoghan1232 — APPROVED** [review]
lgtm ;)

**Pull request merged**